5345 program integrity audit frontend

elipe17 opened the pull request:
## Summary of Changes
- Refactored reports to support different types of file upload inputs, sections vs quarters
- Refactored submission history to support section vs quarter based submissions
- Updated several form classes/selector classes to be more generic and use them across the code base
- Added a bunch of tests for the new components and some tests for components that didnt have tests to get coverage
- Removed year, quarter, and file type from redux state
- Removed the use of the Search Button on the data files pages. Let the form drive the rendering and remove confusing form validation logic
- Hopefully fixed the collapsing file input forever
- Did not completely integrate FRAReports.jsx with all the new updates because this PR was big enough
Pull request closes #5345

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
2. Try really hard to break the datafiles and fra datafiles page :)

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] All STTs are presented a new radio file type option for the "audit active case data"
- [ ] Frontend header validation (year, quarter, program type) still apply to each quarter based file input
- [ ] When submitted/posted to the backend the section type for this universe data should indicate it as such, e.g. `Active Case Data - Audit`. Screenshot below for what is currently posted to backend.
- [ ] Jest tests implemented
- [ ] Spinoff ticket created to implement the E2E tests
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary
- [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) has been updated with testing steps & relevant media

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

jtimpe committed:
models and schemas

jtimpe committed:
models and schemas

jtimpe committed:
models and schemas

jtimpe committed:
move parsers into dir

jtimpe committed:
move parsers into dir

jtimpe committed:
stub program audit parser

jtimpe committed:
fix validation and model relationships

jtimpe committed:
fix validation and model relationships

jtimpe committed:
fix validation and model relationships

jtimpe committed:
fix validation and model relationships

elipe17 committed:
- Use radio component on Reports page
- Rename UploadReport to FileInputForm since it is more clear

elipe17 committed:
- Use radio component on Reports page
- Rename UploadReport to FileInputForm since it is more clear

elipe17 committed:
- Use radio component on Reports page
- Rename UploadReport to FileInputForm since it is more clear

elipe17 committed:
- Use radio component on Reports page
- Rename UploadReport to FileInputForm since it is more clear

elipe17 committed:
- rename to test file to match repo standard

elipe17 committed:
- rename to test file to match repo standard

elipe17 committed:
- rename to test file to match repo standard

jtimpe committed:
move parser_classes

jtimpe committed:
move parser_classes

jtimpe committed:
add program audit classes

jtimpe committed:
test stub

jtimpe committed:
test stub

elipe17 committed:
- Update tests to use a state that actually has SSP
- Update tests to now handle the File Type radio select to always render

elipe17 committed:
- Update tests to use a state that actually has SSP
- Update tests to now handle the File Type radio select to always render

jtimpe committed:
fix duplicate detection

jtimpe committed:
migrations

jtimpe committed:
add program audit section

jtimpe committed:
add program audit section

elipe17 committed:
- remove print statements

elipe17 committed:
- remove print statements

elipe17 committed:
- Removed all redux state for FY, FQ, and File Type
- Made TANF page reactive
- Removed form fields to allow selections drive visibility as opposed tot he search button

elipe17 committed:
- Removed all redux state for FY, FQ, and File Type
- Made TANF page reactive
- Removed form fields to allow selections drive visibility as opposed tot he search button

elipe17 committed:
- updated tanf reports tests to adhere to new reactive interface and no search button

elipe17 committed:
- updated tanf reports tests to adhere to new reactive interface and no search button

elipe17 committed:
- Remove tests related to removed redux state/action/reducers

elipe17 committed:
- Remove tests related to removed redux state/action/reducers

elipe17 committed:
- Resolve test failures for fileuploadform

elipe17 committed:
- Resolve test failures for fileuploadform

elipe17 committed:
- Fix file upload logic for admin

elipe17 committed:
- Fix file upload logic for admin

jtimpe committed:
cat 1-3 validation

jtimpe committed:
cat 1-3 validation

elipe17 committed:
- Update constructYearOptions to take into account a starting year

elipe17 committed:
- Update constructYearOptions to take into account a starting year

elipe17 committed:
- Add new action/reducer to manage transferring between PIA and TANF
- Made FileUpload a bit more agnostic to "section"
- update utility header validators to take PIA into account
- Add unique file upload forms for Quarter based and Section based uploads
- Moved Explainer to separate jsx
- Split out reports to all for different report classes
- Added context and provider for reports to avoid prop drilling
- Add utility constants and functions for PIA
- Split out submission history to keep separation of concerns

elipe17 committed:
- Add new action/reducer to manage transferring between PIA and TANF
- Made FileUpload a bit more agnostic to "section"
- update utility header validators to take PIA into account
- Add unique file upload forms for Quarter based and Section based uploads
- Moved Explainer to separate jsx
- Split out reports to all for different report classes
- Added context and provider for reports to avoid prop drilling
- Add utility constants and functions for PIA
- Split out submission history to keep separation of concerns

elipe17 committed:
- Block multiple submission

elipe17 committed:
- Block multiple submission

elipe17 committed:
- Prevent multiple submission of PIA files

elipe17 committed:
- Prevent multiple submission of PIA files

elipe17 committed:
- Add key for radio select

elipe17 committed:
- Add key for radio select

elipe17 committed:
- Remove unused component and test files

elipe17 committed:
- Remove unused component and test files

elipe17 committed:
- Add test files for new forms and for the actual upload component

elipe17 committed:
- Add test files for new forms and for the actual upload component

elipe17 committed:
- correct extensions

elipe17 committed:
- correct extensions

elipe17 committed:
- Add tests for new submission history components

elipe17 committed:
- Add tests for new submission history components

elipe17 committed:
Merge branch '5345-program-integrity-audit-frontend' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- resolve conflicts

elipe17 committed:
- resolve conflicts

elipe17 committed:
- remove duplicate migration

elipe17 committed:
- remove duplicate migration

jtimpe committed:
add rpt_month_year month validation

elipe17 committed:
- Add program audit to error report
- Add simple admin interface for ease of use

elipe17 committed:
- Add program audit to error report
- Add simple admin interface for ease of use

elipe17 committed:
- Make quarter sub history case agreggates only
- remove prints

elipe17 committed:
- Make quarter sub history case agreggates only
- remove prints

elipe17 committed:
- Update action to allow null quarter to support program audit

elipe17 committed:
- Update action to allow null quarter to support program audit

elipe17 committed:
- Add missing reprocessed state

elipe17 committed:
- Add missing reprocessed state

elipe17 committed:
- Update to make pia case aggregates
- Use correct functino calls

elipe17 committed:
- Update to make pia case aggregates
- Use correct functino calls

elipe17 committed:
Add missing section/program in func

elipe17 committed:
- fix validation error

elipe17 committed:
- fix validation error

elipe17 committed:
- Linting and fix spacing

elipe17 committed:
- Linting and fix spacing

elipe17 committed:
- Add function to map the returned datafile model's section to the frontend expected section if the filetype is program audit

elipe17 committed:
- Add function to map the returned datafile model's section to the frontend expected section if the filetype is program audit

elipe17 committed:
- remove unused prop
- Update tests to actually render sub components

elipe17 committed:
- remove unused prop
- Update tests to actually render sub components

elipe17 committed:
- Make sure quarter upload still validates the quarter of the file

elipe17 committed:
- Make sure quarter upload still validates the quarter of the file

elipe17 committed:
- move selector functions into context
- create re-useable year and quarter components
- re-structure reports dir

elipe17 committed:
- move selector functions into context
- create re-useable year and quarter components
- re-structure reports dir

elipe17 committed:
- Fix test

elipe17 committed:
- Fix test

elipe17 committed:
- Correct label on section based upload

elipe17 committed:
- Correct label on section based upload

elipe17 committed:
- Update tests

elipe17 committed:
- Update tests

elipe17 committed:
- Simplify test cases

elipe17 committed:
- Simplify test cases

elipe17 committed:
- exclude context from coverage

elipe17 committed:
- exclude context from coverage

elipe17 committed:
- Update to support no longer needing search button

elipe17 committed:
- Update to support no longer needing search button

elipe17 committed:
- Remove console logs
- Update FRAReports to have the same form behavior as Reports
- Update reports context to include necessary FRA state

elipe17 committed:
- Remove console logs
- Update FRAReports to have the same form behavior as Reports
- Update reports context to include necessary FRA state

jtimpe committed:
add admin

jtimpe committed:
fix validator

jtimpe committed:
decouple generate_error and delete_duplicate steps

jtimpe committed:
decouple generate_error and delete_duplicate steps

elipe17 committed:
- Add missing logic for FRA to show modal
- Consolidate clear functionality
- remove unused imports

elipe17 committed:
- Add missing logic for FRA to show modal
- Consolidate clear functionality
- remove unused imports

elipe17 committed:
- Update remove previews to not collapse inputs
- Add contextual error modal toggle to handle different input clearing logic

elipe17 committed:
- Update remove previews to not collapse inputs
- Add contextual error modal toggle to handle different input clearing logic

elipe17 committed:
- Store pending change until user decides what to do

elipe17 committed:
- Store pending change until user decides what to do

elipe17 committed:
- Update radioselect and combobox to preserve selection until state changes
- Add new handler to clear pending changes when needed

elipe17 committed:
- Update radioselect and combobox to preserve selection until state changes
- Add new handler to clear pending changes when needed

elipe17 committed:
- Update tests based on new reactive report page
- Add missing setter for fra files when clearing inputs

elipe17 committed:
- Update tests based on new reactive report page
- Add missing setter for fra files when clearing inputs

elipe17 committed:
- Add test files for components
- ignore utils file

elipe17 committed:
- Add test files for components
- ignore utils file

elipe17 committed:
- Update fixture IDs
- Add small wait for inputs since there isn't a way to wait on the render

elipe17 committed:
- Update fixture IDs
- Add small wait for inputs since there isn't a way to wait on the render

elipe17 committed:
Merge branch '5346-program-audit-backend-2' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- Update so that widget displays integrity audit info

elipe17 committed:
- Update so that widget displays integrity audit info

elipe17 committed:
- Test longer timeout

elipe17 committed:
- Test longer timeout

elipe17 committed:
Merge branch '5346-program-audit-backend-2' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- fix merge conflict

elipe17 committed:
- fix merge conflict

elipe17 committed:
- Update combobox to always display value
- Update stt combo box to not render unless it has options

elipe17 committed:
- Update combobox to always display value
- Update stt combo box to not render unless it has options

elipe17 committed:
Merge branch '5345-program-integrity-audit-frontend' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- resolve merge conflict

elipe17 committed:
- Resolve merge conflicts on data files app

elipe17 committed:
Merge branch '5346-program-audit-backend-3' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- Fixing merge conflicts

elipe17 committed:
- Correct file spacing

elipe17 committed:
- Update logging in parser task and use correct method
- Remove console log
- Pass Active case data as section for PIA and program audit true

elipe17 committed:
- Update admin to show if it is program audit
- Pass program audit boolean to parser factory
- Update frontend to send if the file is a program audit
- Update tests to not worry about program audit section

elipe17 committed:
- Remove unnused constant

elipe17 committed:
- Added tests for coverage

elipe17 committed:
- fix lint

elipe17 committed:
- Update viewset to use program type when querying dfs

elipe17 committed:
- Add logic to fix error where viewing SSP content and then switching to an non ssp stt would cause state issues
- Added tests

elipe17 committed:
- Linting

elipe17 committed:
- Update context to reset year input if the currently selected year isnt available when switching program types
- added tests

elipe17 committed:
- Fix cypress tests to adhere to new section format

elipe17 committed:
- Fix failing test

elipe17 committed:
Merge branch '5346-program-audit-backend-3' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
Merge branch '5346-program-audit-backend-3' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- linting

elipe17 committed:
- Remove accidental inclusion

elipe17 committed:
- Make linting happen first to avoid long waits for simple errors
- Remove import

jtimpe commented on tdrs-frontend/cypress/e2e/common-steps/data_files.js:
```diff
@@ -10,15 +10,15 @@ export const restartAtHomePage = () => {
 
 export const fillFYQ = (fiscal_year, quarter) => {
+  cy.wait(1000)
   cy.get('#reportingYears').should('exist').select(fiscal_year)
```
in general, i prefer expanding the timeout on an assertion step so the test doesn't wait longer than it has to. doesn't work in all scenarios though

```suggestion
  cy.get('#reportingYears', { timeout: 10000 }).should('exist').select(fiscal_year)
```

jtimpe commented on tdrs-frontend/cypress/e2e/common-steps/data_files.js:
```diff
@@ -10,15 +10,15 @@ export const restartAtHomePage = () => {
 export const uploadFile = (file_input, file_path, willError = false) => {
   cy.get(file_input).selectFile(file_path, { action: 'drag-drop' })
-  cy.wait(100)
+  cy.wait(1000)
```
would add an assertion on a component being visible/ready as suggested above (if possible)

jtimpe commented on tdrs-frontend/src/actions/reports.js:
```diff
@@ -141,6 +153,16 @@ export const upload =
+// returned to the frontend for the submittedFile's section. However, the frontend maps the Program Audit file's
+// section to be the quarter labels. Thus, we need this function to map the backend to the frontend.
+const map_section = (fileType, submittedFile) => {
+  console.log(fileType)
```
```suggestion

```

jtimpe replied:
additionally, i don't like modifying a function's parameters. prefer to copy, modify, and return the new object.

jtimpe commented on tdrs-frontend/src/components/Reports/ReportsContext.jsx:
```diff
@@ -0,0 +1,253 @@
+      if (sttObject && fileTypeInputValue === 'ssp-moe' && !sttObject.ssp) {
+        setFileTypeInputValue('tanf')
+        dispatch(clearFileList({ fileType: 'tanf' }))
+        dispatch(reinitializeSubmittedFiles('tanf'))
```
not sure it's here only/specifically, but changing STTs while having the Submission History tab selected doesn't trigger a reload of the submission history, making it appear as if there are no submissions for a given search. the old page kicks off a reload after clicking search. another approach is maybe to reset to the file inputs (submission history will load whenever the tab is selected), but I like the reactivity more. maybe a loading spinner (future ticket) would add to the perception that something is happening on the search as well

elipe17 replied:
Yes, I noticed that too. The reason I didn't initially fix it is because that was how the old page behaved until you clicked "Search". I will look at how much more complexity/time is introduced to add a spinner and re-load the submission history when the form inputs change.

jtimpe commented on tdrs-frontend/src/components/FileUploadForms/QuarterFileUploadForm.jsx:
```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useRef } from 'react'
```
this seems to share a lot with `SectionFileUpload` - the differences are mainly the actual section names when rendering and selecting from the redux store. could we use something analogous to `map_section` from `actions/reports.js` and share more of the component structure? i didn't look too closely so there may be more i'm missing

jtimpe commented on tdrs-frontend/src/components/SubmissionHistory/QuarterSubmissionHistory.jsx:
```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react'
```
same comment as the `QuarterFileUploadForm`

elipe17 replied:
I made a lot of updates. I created a few hooks to share the logic between the upload and submission history forms. I also adding reactive loading for the submission history. Now if you change a form input while viewing the submission history, the new history will be automatically loaded and a spinner in each section will render to indicate loading until the new submission history is loaded.

<img width="546" height="612" alt="Screenshot 2025-10-17 at 10 10 03 AM" src="https://github.com/user-attachments/assets/c4ad79d8-2998-4c47-b2d6-e25c413733b1" />

jtimpe reviewed (commented):
Looks and works really well overall! And again, excellent use of context. I'm still testing, just wanted to get a few of my major comments down.

elipe17 committed:
- Updated to use timeout instead of static wait

elipe17 committed:
- Update function to copy and return

elipe17 committed:
- Added new hooks to share logic between file upload components and the submission history components
- Added paginated history to abstract the submission history table for the submission history component
- Re-organized the submission history directory

elipe17 committed:
- Remove bottom border that was accidentally added
- Update submission history hook to have a loading state to manage form changes to reload history for new selections
- Update paginated history to display a loading spinner when the submission history is loading
- Update reducers to handle new loading redux state

elipe17 committed:
- Update tests due to new redux state

elipe17 committed:
- Linting

elipe17 commented on tdrs-frontend/src/components/SubmissionHistory/components/PaginatedHistory.jsx:
```diff
@@ -0,0 +1,57 @@
+  files,
+  loading,
+  reprocessedState,
+  TableComponent,
```
Use children prop?

elipe17 committed:
Merge branch '5346-program-audit-backend-3' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

jtimpe reviewed (approved):
Looks good, works great! Definitely appreciate the context and component splitting

elipe17 committed:
Merge branch '5346-program-audit-backend-3' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
Merge branch '5346-program-audit-backend-3' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 commented:
@mattcoleanderson & @raftmsohani please prioritize this review. The deadline for this work is coming up.

elipe17 committed:
- remove redundent test

raftmsohani reviewed (approved):
tried to break it as suggested, couldn't :( :) Great work

victoriaatraft commented:
@elipe17 Also could not break!

For TANF and FRA Data Files Tabs:
- If a required field is skipped, can we have the error appear to select said skipped field please? For example, on the TANF Data Files Tab, if I selected a FQ but not a FY, the error to select a FQ should appear. Otherwise, I'm left wondering why the content won't load.
- I'm unable to download the first error reports on submission history table. All others after work, but it's files that I've submitted

The rest are minor things I came across testing:
- Can we change Audit Reporting to `Audit Reporting Guidelines` to be consistent with the TANF table
- For Selecting PIA, This Banner that shows `For Additional guidance please refer to the Program Instruction for this new reporting requirement.` comes across to me as an error and it's placement is weird/off looking lol was this an ask from the client or could this be part of the body of the text?
- Minor, but the `Need Help` link for this specific error of uploading a file for the wrong FY:  `File contains data from Oct 1 - Dec 31, which belongs to Fiscal Year 2021, Quarter 1. Adjust your search parameters or upload a different file. Need help? `is different blue than the others.
- There is one error state that says when trying to upload on the TANF `File may correspond to null instead of TANF. Please verify the file type.` Not sure what that means, but we should revisit the language :) 
- Is there a reason why `Work Outcomes of TANF Exiters Submission History` stacked? I'll attach a screenshot 
- The `Download Report Template` link on the FRA Data Files Tab doesn't have a selected state/turns purple after selecting

<img width="1445" height="632" alt="weird stacking of text" src="https://github.com/user-attachments/assets/de3a51ac-56d2-49de-8085-c5f9b69a4af9" />
<img width="1552" height="957" alt="information error banner weird placement" src="https://github.com/user-attachments/assets/17235ba8-991c-4684-8b11-196b028003dd" />
<img width="1562" height="610" alt="different link blues" src="https://github.com/user-attachments/assets/603b0999-8297-41cd-bec0-901745b3694c" />
<img width="1510" height="954" alt="display required error text" src="https://github.com/user-attachments/assets/e4b029e3-9423-4064-aff7-c735a886c006" />

elipe17 committed:
- Update caption

elipe17 committed:
- Add missing class to link

elipe17 commented:
> @elipe17 Also could not break!
> 
> For TANF and FRA Data Files Tabs:
> 
> * If a required field is skipped, can we have the error appear to select said skipped field please? For example, on the TANF Data Files Tab, if I selected a FQ but not a FY, the error to select a FQ should appear. Otherwise, I'm left wondering why the content won't load.
> * I'm unable to download the first error reports on submission history table. All others after work, but it's files that I've submitted
> 
> The rest are minor things I came across testing:
> 
> * Can we change Audit Reporting to `Audit Reporting Guidelines` to be consistent with the TANF table
> * For Selecting PIA, This Banner that shows `For Additional guidance please refer to the Program Instruction for this new reporting requirement.` comes across to me as an error and it's placement is weird/off looking lol was this an ask from the client or could this be part of the body of the text?
> * Minor, but the `Need Help` link for this specific error of uploading a file for the wrong FY:  `File contains data from Oct 1 - Dec 31, which belongs to Fiscal Year 2021, Quarter 1. Adjust your search parameters or upload a different file. Need help? `is different blue than the others.
> * There is one error state that says when trying to upload on the TANF `File may correspond to null instead of TANF. Please verify the file type.` Not sure what that means, but we should revisit the language :)
> * Is there a reason why `Work Outcomes of TANF Exiters Submission History` stacked? I'll attach a screenshot
> * The `Download Report Template` link on the FRA Data Files Tab doesn't have a selected state/turns purple after selecting
> 
> <img alt="weird stacking of text" width="1445" height="632" src="https://private-user-images.githubusercontent.com/106103383/506883162-de3a51ac-56d2-49de-8085-c5f9b69a4af9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE3NDM1MjMsIm5iZiI6MTc2MTc0MzIyMywicGF0aCI6Ii8xMDYxMDMzODMvNTA2ODgzMTYyLWRlM2E1MWFjLTU2ZDItNDlkZS04MDg1LWM1ZjliNjlhNGFmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyOVQxMzA3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MTU4YzJjZjk3YzU1NGQ1ZTNiMDA3ZTM1ZmUzYTZmMmUxNjA0NTdiOWViNWZkMGYyYWJlNjg4YTdkZGE3MjVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.fHMvO0_hL2sFpHQd-extW24L-mDercRRCzuaQJ5xgdk"> <img alt="information error banner weird placement" width="1552" height="957" src="https://private-user-images.githubusercontent.com/106103383/506883173-17235ba8-991c-4684-8b11-196b028003dd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE3NDM1MjMsIm5iZiI6MTc2MTc0MzIyMywicGF0aCI6Ii8xMDYxMDMzODMvNTA2ODgzMTczLTE3MjM1YmE4LTk5MWMtNDY4NC04YjExLTE5NmIwMjgwMDNkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyOVQxMzA3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDU4NTM2YjM5NWEyMGNmMWM4OGQwNzI2MzJhMDZlNzAyMWU3NWFlM2IxNmI5YmYxMTVlMTRhZmI3MDNlMDc5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.i3-WzUjrJSY1pm0lIXcuYR6jYL3qHRWsPO-RpqmJpqk"> <img alt="different link blues" width="1562" height="610" src="https://private-user-images.githubusercontent.com/106103383/506883179-603b0999-8297-41cd-bec0-901745b3694c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE3NDM1MjMsIm5iZiI6MTc2MTc0MzIyMywicGF0aCI6Ii8xMDYxMDMzODMvNTA2ODgzMTc5LTYwM2IwOTk5LTgyOTctNDFjZC1iZWMwLTkwMTc0NWIzNjk0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyOVQxMzA3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wODQ5Yzc4M2UyMzllOTE4NWEwY2EyZDQyYjEyZjI2NmUwYTFjYTg1NDY1MWYzYTFkYjVlZTIwZjRmNzQ1ZjE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.BI-Kdqy0QpJgZoTjzK33A5Kk82UyXv1sBAz0HgiJLT4"> <img alt="display required error text" width="1510" height="954" src="https://private-user-images.githubusercontent.com/106103383/506883185-e4b029e3-9423-4064-aff7-c735a886c006.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE3NDM1MjMsIm5iZiI6MTc2MTc0MzIyMywicGF0aCI6Ii8xMDYxMDMzODMvNTA2ODgzMTg1LWU0YjAyOWUzLTk0MjMtNDA2NC1hZmY3LWM3MzVhODg2YzAwNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyOVQxMzA3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MDYwOGNhYzUxNWU4MzE5NTBmZjkyNDUwZThkOTcwOTc1MzM5NDRhYWM2OGI1M2RlMDczOTNiYjA1NDFjOGI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XT77nLgVxcJnt2BVrY5phgZTRVufV4Km1VuIf9pCV4I">

@victoriaatraft see my responses below.

`If a required field is skipped, can we have the error appear to select said skipped field please? For example, on the TANF Data Files Tab, if I selected a FQ but not a FY, the error to select a FQ should appear. Otherwise, I'm left wondering why the content won't load.`
- Because we removed the `Search` button there is no way to determine an error state. If we implemented form errors without the search button then the errors would always display no matter what until the user made all the appropriate selections. The lack of content is now the form error. Open to coming up with a different approach to guide users.

` I'm unable to download the first error reports on submission history table. All others after work, but it's files that I've submitted`
- Can you give me all the form parameters you used so I can test? I can't reproduce this issue with Program Integrity Audit locally

`Can we change Audit Reporting to "Audit Reporting Guidelines" to be consistent with the TANF table`
- Changed!

` For Selecting PIA, This Banner that shows "For Additional guidance please refer to the Program Instruction for this new reporting requirement." comes across to me as an error and it's placement is weird/off looking lol was this an ask from the client or could this be part of the body of the text?`
- Ah yes, this one is a pain. Because the banner and the explainer are stored in the same row as the Fiscal Year selector the banner is aligned with the top of the fiscal year selector. It looks better if we remove the banner, more similar to the tanf explainer. Open to suggestions on this one too!

`Minor, but the "Need Help" link for this specific error of uploading a file for the wrong FY:  "File contains data from Oct 1 - Dec 31, which belongs to Fiscal Year 2021, Quarter 1. Adjust your search parameters or upload a different file. Need help? "is different blue than the others.`
- Fixed the link color by adding the correct class to the anchor tag

`There is one error state that says when trying to upload on the "TANF File may correspond to null instead of TANF. Please verify the file type." Not sure what that means, but we should revisit the language :)`
- Do you remember what kind of file you tried uploading and what form selections you had?

`Is there a reason why `Work Outcomes of TANF Exiters Submission History` stacked? I'll attach a screenshot`
- That didn't change in this PR. If you go look at the staging apps you will see the same behavior/structure. Open to new ideas to make it pretty :).

`The Download Report Template link on the FRA Data Files Tab doesn't have a selected state/turns purple after selecting`
- I am not sure why the `Download Report Template` link doesn't turn purple after visiting/downloading it. It is configured exactly the same as all other anchor tags. It might be some cross origin thing.

elipe17 committed:
- Fix feed back widget rendering wrong name and widgetId when submitting PIA

elipe17 committed:
- Linting

elipe17 committed:
- Re add form validation logic for the reports page

elipe17 committed:
- Add tests for the form validation states

elipe17 committed:
- linting

elipe17 committed:
- Remove banner from explainer and place it over segmented control

jtimpe commented on tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx:
```diff
@@ -6,8 +6,8 @@ import { getAvailableFileList } from '../../actions/reports'
+import { TotalAggregatesTable } from './components/TotalAggregatesTable'
 
 const SectionSubmissionHistory = ({
   section,
```
may be mistaken, but i don't believe any components in this file are used anymore

elipe17 replied:
They are still used in that file on line 27. But this whole component is actually moot. I just forgot to delete it haha. Thank you for catching

elipe17 committed:
- Update form validation so that it triggers if form fields are not filled in sequentially
- Remove old submission history component

elipe17 committed:
- Update sequential validation logic
- Add tests for the sequential. logic

elipe17 committed:
- Add tests for coverage

elipe17 committed:
Merge branch 'develop' into 5345-program-integrity-audit-frontend

elipe17 committed:
- Cleanup comments

elipe17 committed:
Merge branch '5345-program-integrity-audit-frontend' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 5345-program-integrity-audit-frontend

elipe17 committed:
- Handle edge case for a null program type in a file header
- Add test

elipe17 committed:
- Add missing *

elipe17 committed:
- Update test

victoriaatraft reviewed (approved):
Pending error form logic from Miles, everything else LGTM

elipe17 committed:
Merge branch 'develop' into 5345-program-integrity-audit-frontend

elipe17 committed:
- Place ComboBox in formgroup div instead of react fragment to adhere to correct form error behavior
